sam/models/TimeseriesMLP.py

```diff
@@ -206,64 +109,40 @@ class SamQuantileMLP(BaseTimeseriesRegressor):
-    >>> # easy to predict the weather 1 hour ahead, so the model should do even better.
-    >>> model = SamQuantileMLP(predict_ahead=1, use_y_as_feature=True, timecol='TIME',
-    ...                        quantiles=(0.25, 0.75), epochs=5, verbose=0,
-    ...                        time_components=('hour', 'month', 'weekday'),
-    ...                        time_cyclicals=('hour', 'month', 'weekday'),
-    ...                        time_onehots=None,
-    ...                        rolling_window_size=(1,5,6))
-    >>> # fit returns a keras history callback object, which can be used as normally
-    >>> history = model.fit(X_train, y_train)
-    >>> pred = model.predict(X_test, y_test).dropna()
-    >>> actual = model.get_actual(y_test).dropna()
-    >>> # Because of impossible to know values, some rows have to be dropped. After dropping
-    >>> # them, make sure the indexes still match by dropping the same rows from each side
-    >>> pred, actual = pred.reindex(actual.index).dropna(), actual.reindex(pred.index).dropna()
-    >>> mse_score = mean_squared_error(actual, pred.iloc[:, -1])  # last column contains mean preds
-    >>> print(round(mse_score, 2))
-    66.08
-    >>> # Persistence corresponds to predicting the present, so use ytest
-    >>> persistence_prediction = y_test.reindex(actual.index).dropna()
-    >>> persistence_mse_score = mean_squared_error(actual, persistence_prediction)
-    >>> print(round(persistence_mse_score, 2))
-    149.45
-    >>> # As we can see, the model performs significantly better than the persistence benchmark
-    >>> # Mean benchmark, which does much worse:
-    >>> mean_prediction = pd.Series(y_test.mean(), index = actual)
-    >>> bench_mse_score = mean_squared_error(actual, mean_prediction)
-    >>> print(round(bench_mse_score, 2))
-    2410.59
+    >>> import pandas as pd
+    >>> from sam.models import TimeseriesMLP
+    >>> from sam.feature_engineering import SimpleFeatureEngineer
+    ...
+    >>> data = pd.read_parquet("../data/rainbow_beach.parquet").set_index("TIME")
+    >>> X, y = data, data["water_temperature"]
+    ...
+    >>> simple_features = SimpleFeatureEngineer(
+    >>>     rolling_features=[
+    >>>         ("wave_height", "mean", 24),
+    >>>         ("wave_height", "mean", 12),
+    >>>     ],
+    >>>     time_features=[
+    >>>         ("hour_of_day", "cyclical"),
+    >>>     ],
+    >>>     keep_original=False,
+    >>> )
+    ...
+    >>> model = TimeseriesMLP(
+    >>>     predict_ahead=0,
+    >>>     feature_engineer=simple_features,
+    >>> )
+    ....
+    >>> model.fit(X, y)
     """
 
     def __init__(
         self,
-        predict_ahead: Union[int, Sequence[int]] = 1,
-        quantiles: Sequence[float] = (),
-        use_y_as_feature: bool = True,
-        use_diff_of_y: bool = True,
+        predict_ahead: Union[int, Sequence[int]] = [0],
+        quantiles: Sequence[float] = [],
```

Comment:
Why a list, I changed this to tuples, because they are immutable. We could also use None if we don't want quantiles..